numpyro/distributions/continuous.py

```diff
@@ -1033,6 +1033,115 @@ def icdf(self, q):
         return self.loc + self.scale * logit(q)
 
 
+def _kron(A, B):
+    D = A[..., :, None, :, None] * B[..., None, :, None, :]
+    ds = D.shape
+    newshape = (*ds[:-4], ds[-4] * ds[-3], ds[-2] * ds[-1])
+    return D.reshape(newshape)
+
+
+class MatrixNormal(Distribution):
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "scale_columns": constraints.positive_definite,
+        "scale_rows": constraints.positive_definite,
+    }
+    support = constraints.real_matrix
+    reparametrized_params = ["loc", "scale_columns", "scale_rows"]
+
+    def __init__(self, loc, scale_columns, scale_rows, validate_args=None):
+        """_summary_
+
+        Args:
+            loc (_type_): _description_
+            scale_columns (_type_): _description_
+            scale_rows (_type_): _description_
+            validate_args (_type_, optional): _description_. Defaults to None.
+
+        Raises:
+            ValueError: _description_
+        """
+        if not (loc.ndim == scale_columns.ndim == scale_rows.ndim):
+            raise ValueError(
+                "ndim of loc, scale_columns and scale_rows are reuqired to be equal."
+            )
+
+        event_shape = loc.shape[-2:]
+        n, p = event_shape
+        batch_shape = loc.shape[:-2]
+
+        assert scale_rows.shape == batch_shape + (
+            n,
+            n,
+        ), "scale_rows.shape does not match"
+        assert scale_columns.shape == batch_shape + (
+            p,
+            p,
+        ), "scale_rows.shape does not match"
+
+        super(MatrixNormal, self).__init__(
+            batch_shape=batch_shape, event_shape=event_shape
+        )
+
+        self.loc = loc
+        self.scale_columns = scale_columns
+        self.scale_rows = scale_rows
```

Comment:
I think you can use
```
batch_shape = lax.broadcast_shapes(
    jnp.shape(loc)[:-2], jnp.shape(scale_tril_row)[:-2], jnp.shape(scale_tril_column)[:-2])
self.loc, self.scale_tril_row, self.scale_tril_column = _promote_shapes(loc, scale_tril_row, scale_tril_column, shape=batch_shape)
```

Reply:
Of course. I didn't take care of the case that `loc` and `scale_tril_column`/`scale_tril_row`do not match.

Reply:
That doesn't work. 
```
n=2
p=3
loc = np.random.random(n * p).reshape(n, p)
scale_tril_row=np.random.random((n, n))
scale_tril_row*=np.tri(*scale_tril_row.shape)
scale_tril_column = np.random.random((p, p))
scale_tril_column  *=np.tri(*scale_tril_column.shape)

batch_shape = jax.lax.broadcast_shapes(
    jnp.shape(loc)[:-2],
    jnp.shape(scale_tril_row)[:-2],
    jnp.shape(scale_tril_column)[:-2],
)

promote_shapes(
    loc, scale_tril_row, scale_tril_column, shape=batch_shape
)
```

Reply:
Sorry, I think we need to promote shapes individually. The total shapes of those matrices are not broadcastable. Something like
```
self.loc, = promote_shapes(loc, shape=batch_shape + loc.shape[-2:])
```